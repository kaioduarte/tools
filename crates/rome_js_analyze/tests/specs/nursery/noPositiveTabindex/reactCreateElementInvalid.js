// not working - token_text is adding a space at the end
React.createElement("div", { tabIndex: '1' })
React.createElement("div", { tabIndex: 1 })
