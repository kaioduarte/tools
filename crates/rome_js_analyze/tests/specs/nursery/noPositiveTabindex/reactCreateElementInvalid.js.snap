---
source: crates/rome_js_analyze/tests/spec_tests.rs
assertion_line: 100
expression: reactCreateElementInvalid.js
---
# Input
```js
React.createElement("div", { tabIndex: '1' })
React.createElement("div", { tabIndex: 1 })
React.createElement("div", { tabIndex: +1 })

```

# Diagnostics
```
reactCreateElementInvalid.js:1:30 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
  > 1 │ React.createElement("div", { tabIndex: '1' })
      │                              ^^^^^^^^
    2 │ React.createElement("div", { tabIndex: 1 })
    3 │ React.createElement("div", { tabIndex: +1 })
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
reactCreateElementInvalid.js:2:30 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    1 │ React.createElement("div", { tabIndex: '1' })
  > 2 │ React.createElement("div", { tabIndex: 1 })
      │                              ^^^^^^^^
    3 │ React.createElement("div", { tabIndex: +1 })
    4 │ 
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
reactCreateElementInvalid.js:3:30 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    1 │ React.createElement("div", { tabIndex: '1' })
    2 │ React.createElement("div", { tabIndex: 1 })
  > 3 │ React.createElement("div", { tabIndex: +1 })
      │                              ^^^^^^^^
    4 │ 
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```


